forem/forem
# Hide 'Save' button on posts that belong to the current user

## What type of PR is this? (check all applicable)


## Description
This PR hides the 'Save' button on Posts that belong to the current user.

## Related Tickets & Documents
- Closes https://github.com/forem/forem/issues/16767

## QA Instructions, Screenshots, Recordings
The Bookmark Button for posts appears in multiple areas, so I will do my best to guide you:
1. First, generate a larger number of posts locally, so that you have enough to scroll through and trigger the `buildArticleHTML` function. Run these 4 commands in Rails Console (or save them in a file and call from console):
```ruby
# first command
num_articles = 100

# second command
user_ids = User.all.ids

# third command
public_categories = %w[like unicorn]

# fourth (multi-line) command
num_articles.times do |i|
  tags = []
  tags << "discuss" if (i % 3).zero?
  tags.concat Tag.order(Arel.sql("RANDOM()")).limit(3).pluck(:name)

  markdown = <<~MARKDOWN
    ---
    title:  #{Faker::Book.title} #{Faker::Lorem.sentence(word_count: 2).chomp('.')}
    published: true
    cover_image: #{Faker::Company.logo}
    tags: #{tags.join(', ')}
    ---

    #{Faker::Hipster.paragraph(sentence_count: 2)}
    #{Faker::Markdown.random}
    #{Faker::Hipster.paragraph(sentence_count: 2)}
  MARKDOWN

  article = Article.create!(
    body_markdown: markdown,
    featured: true,
    show_comments: true,
    user_id: User.order(Arel.sql("RANDOM()")).first.id,
  )

  Random.random_number(10).times do |_t|
    article.reactions.create(
      user_id: user_ids.sample,
      category: public_categories.sample,
    )
  end

  article.sync_reactions_count
end
```
2. Check the following scenarios while signed in, and ensure that the `Save` button is hidden on ALL your posts:
a. on the Home Feed (and all other feed tabs), as you scroll down.
b. On your profile page.
c. On the Tag pages (e.g. `localhost:3000/t/discuss`).
d. Create a Series with a few posts. On your Series page (e.g. `localhost:3000/<your-username>/series/1`), none of your series posts have the `Save` button.
e. Create a post under an Organization. On the Organization profile page, none of your posts should have the `Save` button.
f. Search for a phrase in a post such that one of your posts (and preferably another author's) shows up in the results. On your post, there should be no `Save` button. 
3. Check the same scenarios above while signed OUT (at least those which are applicable). Your posts SHOULD have the `Save` button because no user is signed in.

### UI accessibility concerns?
None; no UI changes

## Added/updated tests?

## [Forem core team only] How will this change be communicated?
